### PR TITLE
Post merge clean up for materialized view column comment

### PR DESCRIPTION
**Pull request by vlad-lyutenko**
Some clean ups for this PR https://github.com/trinodb/trino/pull/18016
Correct method order, test adjustments
refactor test access control MV column comment to use mock connector
remove blackhole MV support

## Release notes

( ) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
(x) Release notes are required, with the following suggested text:

```markdown
# BlackHole
* Remove materialized view support 
```

### CODE REVIEW

**Comment by findepi** [comment]
thanks!

**Comment by findepi** [comment]
please update 423 release notes too

**Comment by findepi** [comment]
> please update 423 release notes too

i have other fixes tbd, so will take care of this -- https://github.com/trinodb/trino/pull/18646

File: core/trino-main/src/main/java/io/trino/metadata/Metadata.java [hunk 1]
```diff
@@ -236,11 +236,6 @@ Optional<TableExecuteHandle> getTableHandleForExecute(
      */
     void setViewColumnComment(Session session, QualifiedObjectName viewName, String columnName, Optional<String> comment);
 
-    /**
```
Comment (Praveen2112) [inline core/trino-main/src/main/java/io/trino/metadata/Metadata.java:239]: Can we reword the commit as `Rearrage  setMaterializedViewColumnComment method order`. Can we extract the test changes as a separate commit ?

File: testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java [hunk 2]
```diff
@@ -627,9 +627,9 @@ public void testCommentColumnView()
     public void testCommentColumnMaterializedView()
     {
         String viewName = "comment_materialized_view" + randomNameSuffix();
-        assertUpdate("CREATE MATERIALIZED VIEW " + viewName + " AS SELECT * FROM orders");
-        assertAccessDenied("COMMENT ON COLUMN " + viewName + ".orderkey IS 'new order key comment'", "Cannot comment column to .*", privilege(viewName, COMMENT_COLUMN));
-        assertUpdate(getSession(), "COMMENT ON COLUMN " + viewName + ".orderkey IS 'new comment'");
+        assertUpdate("CREATE MATERIALIZED VIEW mock.default." + viewName + " AS SELECT * FROM orders");
```
Comment (Praveen2112) [inline testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java:630]: Can we specify the operation type on commit message - commit title looks a bit generic

File: plugin/trino-blackhole/src/main/java/io/trino/plugin/blackhole/BlackHoleMetadata.java [hunk 3]
```diff
@@ -24,7 +24,6 @@
 import io.trino.spi.connector.ColumnMetadata;
 import io.trino.spi.connector.ConnectorAnalyzeMetadata;
 import io.trino.spi.connector.ConnectorInsertTableHandle;
-import io.trino.spi.connector.ConnectorMaterializedViewDefinition;
```
Comment (Praveen2112) [inline plugin/trino-blackhole/src/main/java/io/trino/plugin/blackhole/BlackHoleMetadata.java:27]: Do we need to update on the docs side ?

File: core/trino-main/src/main/java/io/trino/metadata/Metadata.java (hunk 1 above)
Comment (vlad-lyutenko) [reply core/trino-main/src/main/java/io/trino/metadata/Metadata.java:239]: refactored

File: testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java (hunk 2 above)
Comment (vlad-lyutenko) [reply testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java:630]: refactored

File: plugin/trino-blackhole/src/main/java/io/trino/plugin/blackhole/BlackHoleMetadata.java (hunk 3 above)
Comment (vlad-lyutenko) [reply plugin/trino-blackhole/src/main/java/io/trino/plugin/blackhole/BlackHoleMetadata.java:27]: updated

File: testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java (hunk 2 above)
Comment (Praveen2112) [reply testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java:630]: Can we rename the commit message as `mock connector` instead of `mock catalog`.

File: docs/src/main/sphinx/connector/blackhole.md [hunk 4]
```diff
@@ -124,9 +124,6 @@ additional features:
 - {doc}`/sql/create-view`
 - {doc}`/sql/show-create-view`
 - {doc}`/sql/drop-view`
-- {doc}`/sql/create-materialized-view`
-- {doc}`/sql/show-create-materialized-view`
-- {doc}`/sql/drop-materialized-view`
```
Comment (Praveen2112) [inline docs/src/main/sphinx/connector/blackhole.md:129]: What if we could squash up with the previous commit ?

**Review by Praveen2112 — APPROVED** [review]

**Review by atanasenko — APPROVED** [review]

**Review by marcinsbd — APPROVED** [review]
